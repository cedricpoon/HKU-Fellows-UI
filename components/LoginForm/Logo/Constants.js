export const SIZE = 150;
