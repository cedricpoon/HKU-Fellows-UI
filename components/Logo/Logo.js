import React, { PureComponent } from 'react';
import Svg,{ Path, G, Line, Polyline, Ellipse } from 'react-native-svg';
import PropTypes from 'prop-types';

class Logo extends PureComponent {
  render() {
    const { size, ...restProps } = this.props;

    return (
      <Svg width={size} height={size} viewBox="0 0 500 512" {...restProps}>
        <Path fill="#4BBFB3" d="M475.866,365.865C491.338,332.573,500,295.323,500,256C500,114.615,388.071,0,250,0S0,114.615,0,256
          s111.929,256,250,256c32.332,0,63.229-6.285,91.593-17.727L475.866,365.865z"/>
        <Path fill="#248477" d="M432.7,366.322c-57.858,11.609-96.37,67.233-88.879,126.269l131.787-126.269
          C461.841,363.534,447.324,363.388,432.7,366.322z"/>
        <G>
          <Line fill="none" stroke="#FFFFFF" strokeWidth="7.68" strokeLinecap="round" strokeLinejoin="round" strokeMiterlimit="10" strokeDasharray="15.36,15.36" x1="119.352" y1="242.474" x2="118.936" y2="354.543"/>
          <Ellipse fill="#FFFFFF" stroke="#FFFFFF" strokeWidth="5.12" strokeLinecap="round" strokeLinejoin="round" strokeMiterlimit="10" cx="118.936" cy="342.021" rx="8.606" ry="8.813"/>
          <Path fill="none" stroke="#FFFFFF" strokeWidth="15" strokeLinecap="round" strokeLinejoin="round" strokeMiterlimit="10" d="
            M408.691,219.362L250,157.457L91.309,219.362c0,0,158.188,58.81,159.195,59.841c0.288,0.295,13.314-4.46,31.774-11.401
            l-0.048,0.019l38.3,19.637v-34.175C362.855,237.113,408.691,219.362,408.691,219.362z"/>
          <Polyline fill="none" stroke="#FFFFFF" strokeWidth="15" strokeLinecap="round" strokeLinejoin="round" strokeMiterlimit="10" points="
            154.281,247.733 154.281,299.118 252.078,337.293 345.719,300.15 345.719,247.733 	"/>
        </G>
      </Svg>
    );
  }
}

Logo.propTypes = {
  size: PropTypes.number.isRequired
}

export default Logo;
