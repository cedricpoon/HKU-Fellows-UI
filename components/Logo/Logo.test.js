import React from 'react';
import { shallow } from 'enzyme';
import Logo from './Logo';

describe('Testing Logo Component', () => {
  const requiredProps = {
    size: 150
  };

  it('renders as expected', () => {
    const wrapper = shallow(
      <Logo {...requiredProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
