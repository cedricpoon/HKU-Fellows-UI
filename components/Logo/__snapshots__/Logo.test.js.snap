// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Logo Component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logo
    size={150}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Path
          d="M475.866,365.865C491.338,332.573,500,295.323,500,256C500,114.615,388.071,0,250,0S0,114.615,0,256 s111.929,256,250,256c32.332,0,63.229-6.285,91.593-17.727L475.866,365.865z"
          fill="#4BBFB3"
        />,
        <Path
          d="M432.7,366.322c-57.858,11.609-96.37,67.233-88.879,126.269l131.787-126.269 C461.841,363.534,447.324,363.388,432.7,366.322z"
          fill="#248477"
        />,
        <G>
          <Line
            fill="none"
            stroke="#FFFFFF"
            strokeDasharray="15.36,15.36"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="7.68"
            x1="119.352"
            x2="118.936"
            y1="242.474"
            y2="354.543"
          />
          <Ellipse
            cx="118.936"
            cy="342.021"
            fill="#FFFFFF"
            rx="8.606"
            ry="8.813"
            stroke="#FFFFFF"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="5.12"
          />
          <Path
            d=" M408.691,219.362L250,157.457L91.309,219.362c0,0,158.188,58.81,159.195,59.841c0.288,0.295,13.314-4.46,31.774-11.401 l-0.048,0.019l38.3,19.637v-34.175C362.855,237.113,408.691,219.362,408.691,219.362z"
            fill="none"
            stroke="#FFFFFF"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="15"
          />
          <Polyline
            fill="none"
            points=" 154.281,247.733 154.281,299.118 252.078,337.293 345.719,300.15 345.719,247.733 	"
            stroke="#FFFFFF"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="15"
          />
        </G>,
      ],
      "height": 150,
      "preserveAspectRatio": "xMidYMid meet",
      "viewBox": "0 0 500 512",
      "width": 150,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "d": "M475.866,365.865C491.338,332.573,500,295.323,500,256C500,114.615,388.071,0,250,0S0,114.615,0,256 s111.929,256,250,256c32.332,0,63.229-6.285,91.593-17.727L475.866,365.865z",
          "fill": "#4BBFB3",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "d": "M432.7,366.322c-57.858,11.609-96.37,67.233-88.879,126.269l131.787-126.269 C461.841,363.534,447.324,363.388,432.7,366.322z",
          "fill": "#248477",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Line
              fill="none"
              stroke="#FFFFFF"
              strokeDasharray="15.36,15.36"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="7.68"
              x1="119.352"
              x2="118.936"
              y1="242.474"
              y2="354.543"
            />,
            <Ellipse
              cx="118.936"
              cy="342.021"
              fill="#FFFFFF"
              rx="8.606"
              ry="8.813"
              stroke="#FFFFFF"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="5.12"
            />,
            <Path
              d=" M408.691,219.362L250,157.457L91.309,219.362c0,0,158.188,58.81,159.195,59.841c0.288,0.295,13.314-4.46,31.774-11.401 l-0.048,0.019l38.3,19.637v-34.175C362.855,237.113,408.691,219.362,408.691,219.362z"
              fill="none"
              stroke="#FFFFFF"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="15"
            />,
            <Polyline
              fill="none"
              points=" 154.281,247.733 154.281,299.118 252.078,337.293 345.719,300.15 345.719,247.733 	"
              stroke="#FFFFFF"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="15"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "fill": "none",
              "stroke": "#FFFFFF",
              "strokeDasharray": "15.36,15.36",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeMiterlimit": "10",
              "strokeWidth": "7.68",
              "x1": "119.352",
              "x2": "118.936",
              "y1": "242.474",
              "y2": "354.543",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cx": "118.936",
              "cy": "342.021",
              "fill": "#FFFFFF",
              "rx": "8.606",
              "ry": "8.813",
              "stroke": "#FFFFFF",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeMiterlimit": "10",
              "strokeWidth": "5.12",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "d": " M408.691,219.362L250,157.457L91.309,219.362c0,0,158.188,58.81,159.195,59.841c0.288,0.295,13.314-4.46,31.774-11.401 l-0.048,0.019l38.3,19.637v-34.175C362.855,237.113,408.691,219.362,408.691,219.362z",
              "fill": "none",
              "stroke": "#FFFFFF",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeMiterlimit": "10",
              "strokeWidth": "15",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "fill": "none",
              "points": " 154.281,247.733 154.281,299.118 252.078,337.293 345.719,300.15 345.719,247.733 	",
              "stroke": "#FFFFFF",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeMiterlimit": "10",
              "strokeWidth": "15",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Path
            d="M475.866,365.865C491.338,332.573,500,295.323,500,256C500,114.615,388.071,0,250,0S0,114.615,0,256 s111.929,256,250,256c32.332,0,63.229-6.285,91.593-17.727L475.866,365.865z"
            fill="#4BBFB3"
          />,
          <Path
            d="M432.7,366.322c-57.858,11.609-96.37,67.233-88.879,126.269l131.787-126.269 C461.841,363.534,447.324,363.388,432.7,366.322z"
            fill="#248477"
          />,
          <G>
            <Line
              fill="none"
              stroke="#FFFFFF"
              strokeDasharray="15.36,15.36"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="7.68"
              x1="119.352"
              x2="118.936"
              y1="242.474"
              y2="354.543"
            />
            <Ellipse
              cx="118.936"
              cy="342.021"
              fill="#FFFFFF"
              rx="8.606"
              ry="8.813"
              stroke="#FFFFFF"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="5.12"
            />
            <Path
              d=" M408.691,219.362L250,157.457L91.309,219.362c0,0,158.188,58.81,159.195,59.841c0.288,0.295,13.314-4.46,31.774-11.401 l-0.048,0.019l38.3,19.637v-34.175C362.855,237.113,408.691,219.362,408.691,219.362z"
              fill="none"
              stroke="#FFFFFF"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="15"
            />
            <Polyline
              fill="none"
              points=" 154.281,247.733 154.281,299.118 252.078,337.293 345.719,300.15 345.719,247.733 	"
              stroke="#FFFFFF"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeMiterlimit="10"
              strokeWidth="15"
            />
          </G>,
        ],
        "height": 150,
        "preserveAspectRatio": "xMidYMid meet",
        "viewBox": "0 0 500 512",
        "width": 150,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "d": "M475.866,365.865C491.338,332.573,500,295.323,500,256C500,114.615,388.071,0,250,0S0,114.615,0,256 s111.929,256,250,256c32.332,0,63.229-6.285,91.593-17.727L475.866,365.865z",
            "fill": "#4BBFB3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "d": "M432.7,366.322c-57.858,11.609-96.37,67.233-88.879,126.269l131.787-126.269 C461.841,363.534,447.324,363.388,432.7,366.322z",
            "fill": "#248477",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Line
                fill="none"
                stroke="#FFFFFF"
                strokeDasharray="15.36,15.36"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="7.68"
                x1="119.352"
                x2="118.936"
                y1="242.474"
                y2="354.543"
              />,
              <Ellipse
                cx="118.936"
                cy="342.021"
                fill="#FFFFFF"
                rx="8.606"
                ry="8.813"
                stroke="#FFFFFF"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="5.12"
              />,
              <Path
                d=" M408.691,219.362L250,157.457L91.309,219.362c0,0,158.188,58.81,159.195,59.841c0.288,0.295,13.314-4.46,31.774-11.401 l-0.048,0.019l38.3,19.637v-34.175C362.855,237.113,408.691,219.362,408.691,219.362z"
                fill="none"
                stroke="#FFFFFF"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="15"
              />,
              <Polyline
                fill="none"
                points=" 154.281,247.733 154.281,299.118 252.078,337.293 345.719,300.15 345.719,247.733 	"
                stroke="#FFFFFF"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeMiterlimit="10"
                strokeWidth="15"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "fill": "none",
                "stroke": "#FFFFFF",
                "strokeDasharray": "15.36,15.36",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeMiterlimit": "10",
                "strokeWidth": "7.68",
                "x1": "119.352",
                "x2": "118.936",
                "y1": "242.474",
                "y2": "354.543",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "cx": "118.936",
                "cy": "342.021",
                "fill": "#FFFFFF",
                "rx": "8.606",
                "ry": "8.813",
                "stroke": "#FFFFFF",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeMiterlimit": "10",
                "strokeWidth": "5.12",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "d": " M408.691,219.362L250,157.457L91.309,219.362c0,0,158.188,58.81,159.195,59.841c0.288,0.295,13.314-4.46,31.774-11.401 l-0.048,0.019l38.3,19.637v-34.175C362.855,237.113,408.691,219.362,408.691,219.362z",
                "fill": "none",
                "stroke": "#FFFFFF",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeMiterlimit": "10",
                "strokeWidth": "15",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "fill": "none",
                "points": " 154.281,247.733 154.281,299.118 252.078,337.293 345.719,300.15 345.719,247.733 	",
                "stroke": "#FFFFFF",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeMiterlimit": "10",
                "strokeWidth": "15",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
